src/ethereum_test_specs/static_state/common/common.py

```diff
@@ -53,116 +48,142 @@ def parse_args_from_string_into_array(stream: str, pos: int, delim: str = " "):
+
+        # Prase :yul
+        elif yul_index != -1:
+            option_start = yul_index + len(yul_marker)
+            options: list[str] = []
+            native_yul_options: str = ""
+
+            if self.code_raw[option_start:].lstrip().startswith("{"):
+                # No yul options, proceed to code parsing
+                source_start = option_start
+            else:
+                opt, source_start = parse_args_from_string_into_array(
+                    self.code_raw, option_start + 1
+                )
+                for arg in opt:
+                    if arg == "object" or arg == '"C"':
+                        native_yul_options += arg + " "
+                    else:
+                        options.append(arg)
+
+            with tempfile.NamedTemporaryFile(mode="w+", delete=False) as tmp:
+                tmp.write(native_yul_options + self.code_raw[source_start:])
+                tmp_path = tmp.name
+            compiled_code = compile_yul(
+                source_file=tmp_path,
+                evm_version=options[0] if len(options) >= 1 else None,
+                optimize=options[1] if len(options) >= 2 else None,
+            )[2:]
+
+        # Prase :abi
+        elif abi_index != -1:
+            abi_encoding = self.code_raw[abi_index + len(abi_marker) + 1 :]
+            tokens = abi_encoding.strip().split()
+            abi = tokens[0]
+            function_signature = function_signature_to_4byte_selector(abi)
+            parameter_str = re.sub(r"^\w+", "", abi).strip()
+
+            parameter_types = parameter_str.strip("()").split(",")
+            if len(tokens) > 1:
+                function_parameters = encode(
+                    [parameter_str],
+                    [
+                        [
+                            int(t.lower(), 0) & ((1 << 256) - 1)  # treat big ints as 256bits
+                            if parameter_types[t_index] == "uint"
+                            else int(t.lower(), 0) > 0  # treat positive values as True
+                            if parameter_types[t_index] == "bool"
+                            else False and ValueError("unhandled parameter_types")
+                            for t_index, t in enumerate(tokens[1:])
+                        ]
+                    ],
+                )
+                return function_signature + function_parameters
+            return function_signature
+
+        # Prase plain code 0x
+        elif self.code_raw.lstrip().startswith("0x"):
+            compiled_code = self.code_raw[2:].lower()
+
+        # Prase lllc code
+        elif self.code_raw.lstrip().startswith("{") or self.code_raw.lstrip().startswith("(asm"):
+            with tempfile.NamedTemporaryFile(mode="w+", delete=False) as tmp:
+                tmp.write(self.code_raw)
+                tmp_path = tmp.name
+
+            # using lllc
+            # result = subprocess.run(["lllc", tmp_path], capture_output=True, text=True)
```

Comment:
I think it should be ok to just require this to be installed already.

At the moment, we prepare releases and fill tests using out external runners, and all of our current external runners point to our server that has already `lllc` installed, so it should just work.

Downside would be that we would be unable to add unit tests for state static tests that contain lllc, but I think that should be ok.